pkg/datalayer/datalayer.go

```diff
@@ -186,3 +234,83 @@ func (dl *DataLayer) WriteCommitsForBranch(volumeName, variantName string, commi
 	encoder.Encode(commits)
 	return nil
 }
+
+func (dl *DataLayer) resolveNamedCommitOnBranch(commit, volumeName, variantName string) (CommitId, error) {
+	remainder := commit[len("HEAD"):]
+	var offset int
+	if remainder == strings.Repeat("^", len(remainder)) {
+		offset = len(remainder)
+	} else {
+		return "", fmt.Errorf("Malformed commit identifier %s", commit)
+	}
+	// Read the commit database
+	commits, err := dl.ReadCommitsForBranch(volumeName, variantName)
+	return commits[len(commits)-1-offset].Id, err
+}
+
+func indexOfCommit(commitId CommitId, commits []Commit) int {
```

Comment:
As discussed change so returns an index (int) and an error of instance NotFound????? 
Sniff in the layer above and return